test/getRunnerOSVersion.test.js

```diff
@@ -0,0 +1,97 @@
+const { describe, it, afterEach } = require('node:test')
+const assert = require('node:assert')
+const { readFileSync } = require('node:fs')
+const { join } = require('node:path')
+
+const { getRunnerOSVersion } = require('../src/lib/getRunnerOSVersion')
+
+describe('getRunnerOSVersion', () => {
+  it('throws if no ImageOS env exists', async () => {
```

Comment:
It'd be best if the tests followed the format we have today, even if it's more rudimentary; the reason for this is that not all maintaining this might be well versed in JavaScript and also because for me this is part of a bigger refactor. If you want to, list what you think could be improved (I think you hinted at it in your description) and we can deal with it later.

Reply:
In the meantime, before discussion continues, I'm gonna see about moving the current tests to something more similar to this.

Reply:
Check https://github.com/erlef/setup-beam/pull/328#issuecomment-2832603425.